hooks/node_handlers/base_export_handler.py

```diff
@@ -0,0 +1,644 @@
+            frame_spec = frame_key._resolve_frame_spec(frame_format_string, frame_key.format_spec)
+            glob_path = re.sub(re.escape(frame_spec), r"*", glob_path)
+        return glob_path
+
+    def _resolve_all_versions_from_fields(self, node, fields, template):
+        """
+        Using fields and a shotgun template, get all the versions on disk
+        that relate to the current item.
+
+        :param node: A :class:`hou.Node` instance.
+        :param dict fields: The template fields.
+        :param template: An :class:`sgtk.Template`.
+
+        :rtype: list(int)
+        """
+        versions = set()
+        if fields:
+            if "version" not in fields:
+                fields["version"] = 1
+            if "name" not in fields:
+                return versions
+
+            def repl_version(match):
+                return "{}*{}".format(match.group(1), match.group(3))
+
+            path = template.apply_fields(fields)
+            glob_path = path
+
+            version_key = template.keys.get("version")
+            if version_key:
+                format_spec = version_key.format_spec or ""
+                match = version_key._FORMAT_SPEC_RE.match(format_spec)
+                if match:
+                    pad_char = match.groups()[0]
+                    min_width = int(match.groups()[-1])
+                else:
+                    pad_char = "0"
+                    min_width = 1
+                max_pad_width = min_width - 1
+                padded_version_regex = r"{pad}{{0,{max}}}\d{{1,}}".format(pad=pad_char, max=max_pad_width)
+                regex = r"([/_\.][vV])({})([/_\.])".format(padded_version_regex)
+                glob_path = re.sub(regex, repl_version, glob_path)
+            
+            glob_path = self._get_sequence_glob_path(glob_path, template, fields)
+
+            version_paths = glob.iglob(glob_path)
+            for key, paths in itertools.groupby(version_paths, key=os.path.dirname):
+                version_path = paths.next()
+                fields = template.get_fields(version_path)
+                versions.add(int(fields["version"]))
+            versions = list(versions)
+            versions.sort()
+        return versions
+
+    def _resolve_version(self, all_versions, current):
+        """
+        From a given string, resolve the current version.
+        Either a specified version or the next version in the sequence.
+
+        :param list(int) all_versions: All the existing versions.
+        :param str current: The currently selected version option.
+
+        :rtype: int
+        """
+        if current != self.NEXT_VERSION_STR:
+            resolved = int(current)
+            every_version = all_versions + [max(all_versions or [0]) + 1]
+            if resolved not in every_version:
+                resolved = max(every_version)
+            return resolved
+        return max(all_versions or [0]) + 1
```

Comment:
I think this compacted version should work

```suggestion
        resolved = max(all_versions or [0]) + 1
        if current != self.NEXT_VERSION_STR:
            every_version = all_versions + [resolved]
            current = int(current)
            if current not in every_version:
                resolved = max(every_version)
        return resolved
```